arch/rv32i/src/pmp.rs

```diff
@@ -531,115 +408,115 @@ impl kernel::mpu::MPU for PMPConfig {
                                 );
-                                csr::CSR.pmpaddr0.set((start as u32) >> 2);
+                                csr::CSR.pmpaddr[0].set((start as u32) >> 2);
 
                                 // Set access to end address
-                                csr::CSR.pmpcfg0.set(cfg_val << 8 | csr::CSR.pmpcfg0.get());
-                                csr::CSR.pmpaddr1.set((start as u32 + size as u32) >> 2);
+                                csr::CSR.pmpcfg[0].set(cfg_val << 8 | csr::CSR.pmpcfg[0].get());
+                                csr::CSR.pmpaddr[1].set((start as u32 + size as u32) >> 2);
                             }
                             1 => {
                                 // Disable access up to the start address
-                                csr::CSR.pmpcfg0.modify(
+                                csr::CSR.pmpcfg[0].modify(
                                     csr::pmpconfig::pmpcfg::r2::CLEAR
                                         + csr::pmpconfig::pmpcfg::w2::CLEAR
                                         + csr::pmpconfig::pmpcfg::x2::CLEAR
                                         + csr::pmpconfig::pmpcfg::a2::TOR,
                                 );
-                                csr::CSR.pmpaddr2.set((start as u32) >> 2);
+                                csr::CSR.pmpaddr[2].set((start as u32) >> 2);
 
                                 // Set access to end address
-                                csr::CSR.pmpcfg0.set(cfg_val << 24 | csr::CSR.pmpcfg0.get());
-                                csr::CSR.pmpaddr3.set((start as u32 + size as u32) >> 2);
+                                csr::CSR.pmpcfg[0].set(cfg_val << 24 | csr::CSR.pmpcfg[0].get());
+                                csr::CSR.pmpaddr[3].set((start as u32 + size as u32) >> 2);
                             }
                             2 => {
                                 // Disable access up to the start address
-                                csr::CSR.pmpcfg1.modify(
+                                csr::CSR.pmpcfg[1].modify(
                                     csr::pmpconfig::pmpcfg::r0::CLEAR
                                         + csr::pmpconfig::pmpcfg::w0::CLEAR
                                         + csr::pmpconfig::pmpcfg::x0::CLEAR
                                         + csr::pmpconfig::pmpcfg::a0::TOR,
                                 );
-                                csr::CSR.pmpaddr4.set((start as u32) >> 2);
+                                csr::CSR.pmpaddr[4].set((start as u32) >> 2);
 
                                 // Set access to end address
-                                csr::CSR.pmpcfg1.set(cfg_val << 8 | csr::CSR.pmpcfg0.get());
-                                csr::CSR.pmpaddr5.set((start as u32 + size as u32) >> 2);
+                                csr::CSR.pmpcfg[1].set(cfg_val << 8 | csr::CSR.pmpcfg[0].get());
```

Comment:
should be pmpcfg[1]
please see all places where changes are needed in https://github.com/tock/tock/pull/1932